sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py

```diff
@@ -0,0 +1,216 @@
+
+        :param str conn_str: The connection string of a Service Bus.
+        :keyword str entity_name: Optional entity name, this can be the name of Queue or Topic.
+         It must be specified if the credential is for specific Queue or Topic.
+        :keyword bool logging_enable: Whether to output network trace logs to the logger. Default is `False`.
+        :keyword transport_type: The type of transport protocol that will be used for communicating with
+         the Service Bus service. Default is `TransportType.Amqp`.
+        :paramtype transport_type: ~azure.servicebus.TransportType
+        :keyword dict http_proxy: HTTP proxy settings. This must be a dictionary with the following
+         keys: `'proxy_hostname'` (str value) and `'proxy_port'` (int value).
+         Additionally the following keys may also be present: `'username', 'password'`.
+        :rtype: ~azure.servicebus.ServiceBusClient
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sync_samples/sample_code_servicebus.py
+                :start-after: [START create_sb_client_from_conn_str_sync]
+                :end-before: [END create_sb_client_from_conn_str_sync]
+                :language: python
+                :dedent: 4
+                :caption: Create a new instance of the ServiceBusClient from connection string.
+
+        """
+        host, policy, key, entity_in_conn_str = _parse_conn_str(conn_str)
+        return cls(
+            fully_qualified_namespace=host,
+            entity_name=entity_in_conn_str or kwargs.get("entity_name"),
+            credential=ServiceBusSharedKeyCredential(policy, key),
+            **kwargs
+        )
+
+    def get_queue_sender(self, queue_name, **kwargs):
+        # type: (str, Any) -> ServiceBusSender
+        """Get ServiceBusSender for the specific queue.
+
+        :param str queue_name: The path of specific Service Bus Queue the client connects to.
+        :keyword int retry_total: The total number of attempts to redo a failed operation when an error occurs.
+         Default value is 3.
+        :rtype: ~azure.servicebus.ServiceBusSender
+        :raises: :class:`ServiceBusConnectionError`
+         :class:`ServiceBusAuthorizationError`
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sync_samples/sample_code_servicebus.py
+                :start-after: [START create_servicebus_sender_from_sb_client_sync]
+                :end-before: [END create_servicebus_sender_from_sb_client_sync]
+                :language: python
+                :dedent: 4
+                :caption: Create a new instance of the ServiceBusSender from ServiceBusClient.
+
+        """
+        # pylint: disable=protected-access
+        sender = ServiceBusSender(
+            fully_qualified_namespace=self.fully_qualified_namespace,
+            queue_name=queue_name,
+            credential=self._credential,
+            logging_enable=self._config.logging_enable,
```

Comment:
We should allow people to override this on a per call bases as well:
```
logging_enable=kwargs.pop('logging_enable', self._config.logging_enable)
```

Reply:
If I remember correctly, network tracing happens at the connection level. For separate connection, exposing logging_enable for each call seems reasonable.
But if we apply connection sharing, new uamqp.session/link(SBSender/Receiver) can't touch the network tracing setting on the shared connection or we can do that?

Reply:
Ah - yes I'm pretty sure correct!